packages/components/src/form/index.js

```diff
@@ -84,12 +95,23 @@ class Form extends Component {
 
 	async handleSubmit() {
 		const { values } = this.state;
+		const { onSubmitCallback, onSubmit } = this.props;
 		const touched = {};
 		Object.keys( values ).map( ( name ) => ( touched[ name ] = true ) );
 		this.setState( { touched } );
 
 		if ( await this.isValidForm() ) {
-			this.props.onSubmitCallback( values );
+			const callback = onSubmitCallback ? onSubmitCallback : onSubmit;
```

Comment:
This should also default to `onSubmit` as the `Callback` prop is deprecated

Reply:
sorry, my mistake! The change has been pushed.